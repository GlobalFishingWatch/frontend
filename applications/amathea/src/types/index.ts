import { Feature, Geometry } from 'geojson'

export type WorkspaceParam = 'zoom' | 'latitude' | 'longitude' | 'start' | 'end' | 'modal'

export type ModalTypes = 'newWorkspace' | 'newAOI' | 'newDataset' | 'shareWorkspace'

export type QueryParams = {
  zoom?: number
  latitude?: number
  longitude?: number
  start?: string
  end?: string
  modal?: ModalTypes
}

export type ModalConfigOption = { title: string; component: string }

export type ModalConfigOptions = {
  [query in ModalTypes]?: ModalConfigOption
}

export type MapCoordinates = {
  latitude: number
  longitude: number
  zoom: number
}

export type Editor = { id: string; email: string }

export type WorkspaceConfig = {
  id: string
  label: string
  description?: string
  editors?: Editor[]
}

export type AOIConfig = {
  id: string
  label: string
  geometry: Feature<Geometry, unknown>
}

export type DatasetConfig = {
  id: string
  label: string
  editors?: Editor[]
}
