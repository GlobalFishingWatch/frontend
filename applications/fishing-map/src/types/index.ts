export type WorkspaceParam = 'zoom' | 'latitude' | 'longitude' | 'start' | 'end'

export type QueryParams = {
  [query in WorkspaceParam]?: string | number | boolean | null
}

export type MapCoordinates = {
  latitude: number
  longitude: number
  zoom: number
}
