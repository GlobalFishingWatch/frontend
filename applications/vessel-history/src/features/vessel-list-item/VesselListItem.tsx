import React from 'react'
import Logo from '@globalfishingwatch/ui-components/src/logo'
import IconButton from '@globalfishingwatch/ui-components/src/icon-button/IconButton'
import styles from './VesselListItem.module.css'
import '@globalfishingwatch/ui-components/dist/base.css'

interface LoaderProps {
  saved?: boolean
  vessel: any
}

const VesselListItem: React.FC<LoaderProps> = (props): React.ReactElement => {
  return (
    <div className={styles.vesselItem}>
      {props.saved && (
        <IconButton
          type="default"
          size="default"
          icon="delete"
          className={styles.deleteSaved}
        ></IconButton>
      )}
      <h3>Don Ying Gang 8003</h3>
      <div className={styles.identifiers}>
        <div>
          <label>FLAG</label>
          Chinese Taipei
        </div>
        <div>
          <label>MMSI</label>
          31306526
        </div>
        <div>
          <label>IMO</label>
          12345
        </div>
        <div>
          <label>CALLSIGN</label>
          407201
        </div>
        <div>
          <label>TRANSMISSIONS</label>
          from 2015-07-21 to 2020/04/13
        </div>
        <div>
          <label>IMO</label>
          12345
        </div>
        <div>
          <label>SAVED ON</label>
          2020/08/01
        </div>
      </div>
    </div>
  )
}

export default VesselListItem
