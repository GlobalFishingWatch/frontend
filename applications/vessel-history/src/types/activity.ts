import { EncounterEvent, EventVessel, PointCoordinate } from "@globalfishingwatch/api-types/dist/events";
import { GroupRegions } from "features/regions/regions.slice";

export interface NextPort {
    id: string;
    label: string;
    iso: string;
    anchorageId: string;
}

export enum EventType {
    Fishing = 'fishing',
    Encounter = 'encounter',
}
export interface Regions {
    arg: any[];
    eez: string[];
    fao: string[];
    hsp: any[];
    kkp: any[];
    vme: string[];
    ames: string[];
    rfmo: string[];
    mpant: any[];
    mparu: any[];
    ocean: string[];
    other: any[];
    mregion: string[];
    majorFao: string[];
}

export interface Distances {
    startDistanceFromShoreKm?: number;
    endDistanceFromShoreKm: number;
    startDistanceFromPortKm?: number;
    endDistanceFromPortKm: number;
}

export interface Fishing {
    totalDistanceKm: number;
    averageSpeedKnots: number;
    averageDurationHours: number;
}
export interface RegionAuthorization {
    rfmo: string;
    authorized: boolean;
}
export interface Authorization {
    rfmo: string;
    authorized: boolean;
}
export interface VesselAuthorization {
    id: string;
    authorizations: Authorization[];
}


export interface ActivityEvent {
    type: EventType;
    vessel: EventVessel;
    start: string;
    end: string;
    position: PointCoordinate;
    regions: Regions;
    boundingBox: number[];
    distances: Distances;
    fishing?: Fishing;
    encounter?: EncounterEvent;
}

export interface ActivityFishingEvent extends ActivityEvent {
    fishing: Fishing;
}
export interface ActivityEncounterEvent extends ActivityEvent {
    encounter: EncounterEvent;
}

export type ActivityEventType = ActivityFishingEvent | ActivityEncounterEvent
export interface ActivityEventGroup {
    event_type: EventType
    event_places: GroupRegions[]
    ocean?: string
    start: string
    end: string
    encounter?: EncounterEvent
    open: boolean
    entries: ActivityEventType[];
}

export interface ActivityEvents {
    total: number;
    limit?: any;
    groups: ActivityEventGroup[];
}

export interface OfflineVesselActivity {
    activities: ActivityEventType[]
    profileId: string
    savedOn: string
}
