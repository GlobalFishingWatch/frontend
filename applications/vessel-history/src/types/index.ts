export type WorkspaceParam =
  | 'zoom'
  | 'latitude'
  | 'longitude'
  | 'start'
  | 'end'
  | 'vessel'
  | 'timebarMode'

export type QueryParams = {
  [query in WorkspaceParam]?: string | number | boolean | null
}

export type CoordinatePosition = {
  latitude: number
  longitude: number
}

export type MapCoordinates = {
  latitude: number
  longitude: number
  zoom: number
}

export interface OtherCallsign {
  counter: number
  name: string
}

export interface OtherImo {
  counter: number
  name: string
}

export interface OtherShipname {
  counter: number
  name: string
}

export interface Vessel {
  id: string
  callsign: string
  firstTransmissionDate: Date
  flag: string
  imo?: any
  lastTransmissionDate: Date
  mmsi: string
  otherCallsigns: OtherCallsign[]
  otherImos: OtherImo[]
  otherShipnames: OtherShipname[]
  shipname: string
  source: string
  dataset: string
  vesselMatchId: string
}
