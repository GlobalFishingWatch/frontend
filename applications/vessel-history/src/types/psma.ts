export type Psma = {
  id: string
  country: string
  iso3: string
  signature: string | null
  ratification: string | null
  acceptance: string | null
  approval: string | null
  accession: string | null
}
