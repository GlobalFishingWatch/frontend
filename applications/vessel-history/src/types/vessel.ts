import { VesselSearch } from '@globalfishingwatch/api-types/dist'

export type VesselSourceId = {
  [key: string]: any
}

export interface OfflineVessel extends VesselSearch {
  profileId: string
  savedOn: string
}
