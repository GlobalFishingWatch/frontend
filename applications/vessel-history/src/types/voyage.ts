import { ActivityEvent } from 'types/activity'

export enum EventTypeVoyage {
  Voyage = 'voyage',
}

export interface Voyage {
  from?: ActivityEvent
  to?: ActivityEvent
  type: EventTypeVoyage
  start: number
  end: number
  timestamp: number
}

export interface RenderedVoyage extends Voyage {
  status: 'collapsed' | 'expanded'
  visible: boolean
}
