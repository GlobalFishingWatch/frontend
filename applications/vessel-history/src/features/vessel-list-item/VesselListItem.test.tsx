import { render } from '@testing-library/react'
import { Vessel } from 'types'
import VesselListItem from './VesselListItem'

const mockShowSourceAPI = jest.fn()
jest.mock('data/constants', () => ({
  get SHOW_VESSEL_API_SOURCE() {
    return mockShowSourceAPI() // set some default value
  },
}))
jest.mock('redux-first-router-link')

describe('<VesselListItem />', () => {
  const vessel: Vessel = {
    callsign: 'CB5527',
    firstTransmissionDate: '2012-01-19T14:24:39Z',
    flag: 'CHL',
    id: '9e075a986-6162-fe6e-b25e-81188438a00c',
    imo: null,
    lastTransmissionDate: '2020-12-13T23:51:36Z',
    mmsi: '725000410',
    otherCallsigns: [],
    otherImos: [],
    otherShipnames: [],
    shipname: 'DON TITO',
    source: 'AIS',
    dataset: 'public-global-vessels:v20190502',
    vesselMatchId: '6dd26b05-c055-5b5a-b396-2cc6503fdd4c',
  }

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('shows source API in environments others than production', () => {
    mockShowSourceAPI.mockReturnValueOnce(true)
    const component = render(<VesselListItem vessel={vessel} />)
    expect(component.getAllByText('SOURCE').length).toBe(1)
    expect(component.getAllByText('GFW+TMT').length).toBe(1)
    expect(component.asFragment()).toMatchSnapshot()
  })

  it('does not show source API in production environment', () => {
    mockShowSourceAPI.mockReturnValueOnce(false)
    const component = render(<VesselListItem vessel={vessel} />)
    expect(() => component.getAllByText('SOURCE')).toThrow(
      'Unable to find an element with the text: SOURCE.'
    )
    expect(component.asFragment()).toMatchSnapshot()
  })
})
