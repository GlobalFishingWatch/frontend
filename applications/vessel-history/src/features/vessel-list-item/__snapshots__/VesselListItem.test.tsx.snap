// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VesselListItem /> does not show source API in production environment 1`] = `
<DocumentFragment>
  <div
    class="vesselItem"
  >
    <a
      href="[\\"profile\\",\\"public-global-vessels:v20190502\\",\\"9e075a986-6162-fe6e-b25e-81188438a00c\\",\\"6dd26b05-c055-5b5a-b396-2cc6503fdd4c\\"]"
    >
      <h3>
        DON TITO
      </h3>
    </a>
    <div
      class="identifiers"
    >
      <div>
        <label>
          FLAG
        </label>
        Chile
      </div>
      <div>
        <label>
          MMSI
        </label>
        725000410
      </div>
      <div>
        <label>
          CALLSIGN
        </label>
        CB5527
      </div>
      <div>
        <label>
          TRANSMISSIONS
        </label>
        from 2012-01-19T14:24:39Z to 2020-12-13T23:51:36Z
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<VesselListItem /> shows source API in environments others than production 1`] = `
<DocumentFragment>
  <div
    class="vesselItem"
  >
    <a
      href="[\\"profile\\",\\"public-global-vessels:v20190502\\",\\"9e075a986-6162-fe6e-b25e-81188438a00c\\",\\"6dd26b05-c055-5b5a-b396-2cc6503fdd4c\\"]"
    >
      <h3>
        DON TITO
      </h3>
    </a>
    <div
      class="identifiers"
    >
      <div>
        <label>
          FLAG
        </label>
        Chile
      </div>
      <div>
        <label>
          MMSI
        </label>
        725000410
      </div>
      <div>
        <label>
          CALLSIGN
        </label>
        CB5527
      </div>
      <div>
        <label>
          SOURCE
        </label>
        GFW+TMT
      </div>
      <div>
        <label>
          TRANSMISSIONS
        </label>
        from 2012-01-19T14:24:39Z to 2020-12-13T23:51:36Z
      </div>
    </div>
  </div>
</DocumentFragment>
`;
